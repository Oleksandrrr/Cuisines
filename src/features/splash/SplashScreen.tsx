import React, {useEffect} from 'react';
import {View, Text, StyleSheet, Dimensions} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

const {width, height} = Dimensions.get('window');

interface SplashScreenProps {
  onFinish: () => void;
}

const SplashScreen: React.FC<SplashScreenProps> = ({onFinish}) => {
  useEffect(() => {
    const timer = setTimeout(() => {
      onFinish();
    }, 3000); // Показываем splash screen 3 секунды

    return () => clearTimeout(timer);
  }, [onFinish]);

  return (
    <LinearGradient
      colors={['#9933FF', '#33CCFF']}
      start={{x: 0, y: 1}} // Начинаем с нижнего левого угла
      end={{x: 1, y: 0}} // Заканчиваем в верхнем правом углу
      style={styles.container}>
      <View style={styles.content}>
        <Text style={styles.title}>RaisinEat</Text>
      </View>
    </LinearGradient>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width,
    height,
  },
  content: {
    alignItems: 'center',
  },
  title: {
    fontSize: 48,
    fontWeight: 'bold',
    color: 'white',
    textAlign: 'center',
  },
});

export default SplashScreen;
