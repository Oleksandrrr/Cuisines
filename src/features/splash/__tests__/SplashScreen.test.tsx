import React from 'react';
import {render, waitFor} from '@testing-library/react-native';
import SplashScreen from '../SplashScreen';

// Mock react-native-linear-gradient
jest.mock('react-native-linear-gradient', () => {
  const {View} = require('react-native');
  return View;
});

// Mock Dimensions
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');
  return {
    ...RN,
    Dimensions: {
      get: jest.fn().mockReturnValue({
        width: 375,
        height: 812,
      }),
    },
  };
});

describe('SplashScreen', () => {
  const mockOnFinish = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('renders correctly', () => {
    const {toJSON} = render(<SplashScreen onFinish={mockOnFinish} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onFinish after 3 seconds', async () => {
    render(<SplashScreen onFinish={mockOnFinish} />);

    // Fast-forward time by 3 seconds
    jest.advanceTimersByTime(3000);

    await waitFor(() => {
      expect(mockOnFinish).toHaveBeenCalledTimes(1);
    });
  });

  it('displays RaisinEat text', () => {
    const {getByText} = render(<SplashScreen onFinish={mockOnFinish} />);
    expect(getByText('RaisinEat')).toBeTruthy();
  });

  it('does not call onFinish before 3 seconds', () => {
    render(<SplashScreen onFinish={mockOnFinish} />);

    // Fast-forward time by 2.9 seconds
    jest.advanceTimersByTime(2900);

    expect(mockOnFinish).not.toHaveBeenCalled();
  });
});
