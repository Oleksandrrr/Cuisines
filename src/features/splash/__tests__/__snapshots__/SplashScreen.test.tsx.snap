// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders correctly 1`] = `
<View
  colors={
    [
      "#9933FF",
      "#33CCFF",
    ]
  }
  end={
    {
      "x": 1,
      "y": 0,
    }
  }
  start={
    {
      "x": 0,
      "y": 1,
    }
  }
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 48,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      RaisinEat
    </Text>
  </View>
</View>
`;
